### ISSUE TITLE
Difference of Mobile OS usage(iOS, Android) - 4.1.2: Name, Role, Value

### ISSUE THREAD

**Issue by lgnacio** [w3c/wcag#1339]
(state of VoiceOver/Talkback turned on)In iOS explicitly provides a "Cancel" button to close the layer pop-up, Android does not have an explicit way to close the layer pop-up. However, the user can close the layer pop-up with a gesture (two-finger-tap). If gestures are not considered it should be developer to create a completely custom implementation? Or is gesture a mechanism for cancellation? I think this issue is extension of #1070

iOS:
![C2613CDF-0F30-4863-B3FF-76690020B6CA](https://user-images.githubusercontent.com/7776510/90999129-be944d00-e600-11ea-84ce-1cccfdfaa127.png)
Android:
![Screenshot_20200824-103208](https://user-images.githubusercontent.com/7776510/90999135-c18f3d80-e600-11ea-8aa8-c512472e4858.png)

**Comment by patrickhlauke** [comment]
i think this question mixes up a few concepts. first, i don't think this is relevant to 4.1.2. next, when you say "If gestures are not considered" - assume you're referring to 2.5.1 Pointer Gestures ... but see the note for that SC

> This requirement applies to web content that interprets pointer actions (i.e. this does not apply to actions that are required to operate the user agent or assistive technology).

so any gestures you use to operate the AT are not covered by this. unless you mean that even without AT running, the only way to dismiss is a two-finger tap.

I would however ask if there's a way to dismiss layers with the keyboard. If not, this may well be a failure of 2.1.1. and fundamentally, if you're applying WCAG to native (which I think is the context you're operating in, based on previous questions), then yes - the shortcoming may be part of the standard component that Android or iOS give you, but at the end of the day it's still a barrier/stumbling block to users, so it's a failure regardless. (compared to web content, where a shortcoming in the browser/UA or the OS may not apply to the web *content* per se, since it's not directly under the author's control).

**Comment by lgnacio** [comment]
@patrickhlauke then, Can the mobile "back" button be a mechanism for cancellation?


![image](https://user-images.githubusercontent.com/7776510/91024769-6460af80-e633-11ea-86b5-da801c8c0f77.png)

**Comment by patrickhlauke** [comment]
I'd say so, yes

**Comment by lgnacio** [comment]
@patrickhlauke (when the screen reader turned on)If the layer cannot be canceled, which SC is it? I think it's a different issues with the 2.1.1 keyboard. WCAG regards AT as a keyboard?

**Comment by patrickhlauke** [comment]
splitting this out:

- assuming that when the user is using a keyboard (with or without AT running) they can cancel out, this passes 2.1.1
- if you're saying that there's no way for the user to cancel when running the AT (but without keyboard), then that's a failure that is currently not covered directly by WCAG, i'd say. it would be something that can be mentioned as a best practice / failure that's beyond WCAG itself (and the best way to address this is to suggest some kind of visible/focusable close control that helps all users)

**Comment by lgnacio** [comment]
@patrickhlauke here is example

**Comment by lgnacio** [comment]
> if you're saying that there's no way for the user to cancel when running the AT (but without keyboard), then that's a failure that is currently not covered directly by WCAG, i'd say. 

- Here is an example(Google map app). People use "swipe gesture" or "map space touch" to layer close because there is not "back" button. But blind people cannot use this way. So developer is created a completely custom implementation that layer close for VoiceOver user.

![94CA2213-A023-40F1-97D1-EB85EE58A068](https://user-images.githubusercontent.com/7776510/91114100-0cbd5500-e6c2-11ea-93d0-78298e4881e7.png)

- Most apps are not consider this mechanism. so Blind people are not recognize that relationships(layer and main screen) when the presentation format changes. So is it to fail the 1.3.1 A  SC?

**Comment by patrickhlauke** [comment]
> Here is an example(Google map app)

that looks like a good approach/implementation to me

> Blind people are not recognize that relationships(layer and main screen) when the presentation format changes. So is it to fail the 1.3.1 A SC

I would say it's not necessary for a user to understand or not that they're in a layer over the main screen. at that point, they've jumped into a view that represents a menu. It's irrelevant if, in practice, that menu is "over" the main screen, or if it's screen-filling and effectively a separate screen, or anything else. The user's context at that point is "you're in this menu of options". So I don't see any issue here.
